spec/helpers/set.js

```diff
@@ -0,0 +1,79 @@
+'use babel'
+
+import {setEqual} from '../../src/set-helpers'
+
+Set.prototype.isEqual = (other) => { // eslint-disable-line no-extend-native
+  if (other instanceof Set) {
+    return setEqual(this, other)
+  } else {
+    return undefined
+  }
+}
+
+Set.prototype.jasmineToString = () => { // eslint-disable-line no-extend-native
+  let result = 'Set {'
+  let first = true
+  this.forEach((element) => {
+    if (!first) {
+      result += ', '
+    }
+    result += element.toString()
+    return result
+  })
+  first = false
+  result + '}'
+}
+
+let toEqualSet = (expectedItems, customMessage) => {
+  let pass = true
+  let expectedSet = new Set(expectedItems)
+  if (customMessage == null) {
+    customMessage = ''
+  }
+
+  expectedSet.forEach((item) => {
+    if (!this.actual.has(item)) {
+      pass = false
+      this.message = () => {
+        return 'Expected set ' + (formatSet(this.actual)) + ' to have item ' + item + '. ' + customMessage
+      }
+      return this.message
+    }
+  })
+  this.actual.forEach((item) => {
+    if (!expectedSet.has(item)) {
+      pass = false
+      this.message = () => {
+        return 'Expected set ' + (formatSet(this.actual)) + ' not to have item ' + item + '. ' + customMessage
+      }
+      return this.message
+    }
+  })
+  return pass
+}
+
+let formatSet = (set) => {
+  return '(' + (setToArray(set).join(' ')) + ')'
+}
+
+let setToArray = (set) => {
+  let items = []
+  set.forEach((item) => {
+    return items.push(item)
+  })
+  return items.sort()
+}
+
+// let currentSpecFailed = () => {
+//   console.log(jasmine.getEnv())
+//   return jasmine
+//     .getEnv()
+//     .currentSpec
+//     .results()
+//     .getItems()
+//     .some((item) => {
+//       return !item.passed()
+//     })
+// }
```

Comment:
The main reason we need it is for randomized tests... we want them to terminate as soon as a single failure occurs rather than piling up more iterations. Is there a better approach to getting early termination in those tests?